packages/zoltan2/src/algorithms/partition/Zoltan2_AlgPuLP.hpp

```diff
@@ -342,11 +351,20 @@ void AlgPuLP<Adapter>::partition(
   int* vertex_weights = NULL;
   long vertex_weights_sum = 0;
   if (nVwgts) {
+    nVwgts = 1;
     vertex_weights = new int[nVtx];
-    scale_weights(nVtx, vwgts[0], vertex_weights);
+    scale_weights(nVtx, nVwgts, vwgts, vertex_weights);
     for (int i = 0; i < num_verts; ++i)
       vertex_weights_sum += vertex_weights[i];
   }
+#else
+  // XtraPuLP supports an arbitrary number of vertex weights
+  int* vertex_weights = NULL;
+  if (nVwgts) {
+    vertex_weights = new int[nVtx*nVwgts];
```

Comment:
Use of raw `new` and `delete` is an antipattern.  Prefer `unique_ptr` if you mean to transfer ownership to the callee (`scale_weights`).